beep/helpers/featurizer_helpers.py

```diff
@@ -285,109 +275,218 @@ def get_hppc_ocv_helper(cycle_hppc_0, step_num):
-    step = 11
-    step_later = 43
-    cycle_hppc_0 = cycle_hppc.loc[cycle_hppc.cycle_index == 2]
+    cycles = cycle_hppc.cycle_index.unique()
+
+    cycle_hppc_0 = cycle_hppc.loc[cycle_hppc.cycle_index == cycles[0]]
     #     in case that cycle 2 correspond to two cycles one is real cycle 2, one is at the end
     cycle_hppc_0 = cycle_hppc_0.loc[cycle_hppc_0.test_time < 250000]
+
+    step = 11
+    step_later = 43
+
     voltage_1 = get_hppc_ocv_helper(cycle_hppc_0, step)
-    chosen = cycle_hppc.loc[cycle_hppc.cycle_index == diag_num]
+    chosen = cycle_hppc.loc[cycle_hppc.cycle_index == cycles[diag_pos]]
     voltage_2 = get_hppc_ocv_helper(chosen, step_later)
+
     dv = list_minus(voltage_1, voltage_2)
-    return np.var(dv)
 
+    var_dv = np.var(dv)
+
+    hppc_ocv_features['variance of ocv'] = [var_dv]
+
+    return hppc_ocv_features
+
+
+def get_chosen_df(processed_cycler_run, diag_pos):
+    """
+    This function narrows your data down to a dataframe that contains only the diagnostic cycle number you
+    are interested in.
+
+    Args:
+        processed_cycler_run (beep.structure.ProcessedCyclerRun)
+        diag_pos (int): diagnostic cycle occurence for a specific <diagnostic_cycle_type>. e.g.
+        if rpt_0.2C, occurs at cycle_index = [2, 42, 147, 249 ...], <diag_pos>=0 would correspond to cycle_index 2.
 
-def get_hppc_r(processed_cycler_run, diag_num):
-    '''
-    This function takes in cycling data for one cell and returns the resistance at different SOCs with resistance at the
-    first hppc cycle(cycle 2) deducted
-    Argument:
-            processed_cycler_run(process_cycler_run object)
-            diag_num(int): diagnostic cycle number at which you want to get the feature, such as 37 or 142
     Returns:
-            two floats
-            the variance of the diag_num - cycle 2 for HPPC resistance for both charge and discharge
-    '''
+        a datarame that only has the diagnostic cycle you are interested in, and there is a column called
+        'diagnostic_time[h]' starting from 0 for this dataframe.
+    """
+
     data = processed_cycler_run.diagnostic_interpolated
-    cycle_hppc = data.loc[data.cycle_type == 'hppc']
-    cycle_hppc = cycle_hppc.loc[cycle_hppc.current.notna()]
-    cycles = cycle_hppc.cycle_index.unique()
-    if diag_num not in cycles:
+    hppc_cycle = data.loc[data.cycle_type == 'hppc']
+    hppc_cycle = hppc_cycle.loc[hppc_cycle.current.notna()]
+    cycles = hppc_cycle.cycle_index.unique()
+    diag_num = cycles[diag_pos]
+
+    chosen = hppc_cycle.loc[hppc_cycle.cycle_index == diag_num]
+    chosen = chosen.sort_values(by='test_time')
+    chosen['diagnostic_time'] = (chosen.test_time - chosen.test_time.min()) / 3600
+
+    return chosen
+
+
+def res_calc(chosen, diag_pos, soc, step_ocv, step_cur, index):
+    """
+    This function calculates resistance based on different socs and differnet time scales in hppc cycles.
+
```

Comment:
The docstring for this method and `get_r_df` appear to be identical. I think the latter needs to be changed?